File: LibSWBF2.NET/Logging/LoggerEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.InteropServices;

namespace LibSWBF2.Logging
{
    public enum ELogType : byte
    {
        Info = 0,
        Warning = 1,
        Error = 2
    }

    public class LoggerEntry
    {
        public string Message { get; private set; }
        public ELogType Level { get; private set; }
        public uint Line { get; private set; }
        public string File { get; private set; }

        internal LoggerEntry(string msg, ELogType level, uint line, string file)
        {
            Message = msg;
            Level = level;
            Line = line;
            File = file;
        }

        public override string ToString()
        {
            if (Level == ELogType.Info)
            {
                return "[" + Level + "] " + Message;
            }
            else
            {
                return "[" + Level + "] " + Message + " - IN " + Line + " " + File;
            }
        }
    }
}

File: LibSWBF2/Logging/LoggerEntry.cpp
#include "stdafx.h"
#include "LoggerEntry.h"
#include <string>

namespace LibSWBF2::Logging
{
	LoggerEntry::LoggerEntry(const char* message, const ELogType level, const uint64_t line, const char* file)
	{
		this->m_Message = message;
		this->m_Level = level;
		this->m_Line = line;
		this->m_File = file;
	}

	String LoggerEntry::ToString() const
	{
		if (m_Level == ELogType::Info)
		{
			return "[" + LogTypeToString(m_Level) + "] " + m_Message;
		}
		else
		{
			return "[" + LogTypeToString(m_Level) + "] " + m_Message + " - IN " + std::to_string(m_Line).c_str() + " " + m_File;
		}
	}
}